warp10/src/main/java/io/warp10/script/binary/ComparisonOperation.java

```diff
@@ -186,6 +190,10 @@ public Object comparison(WarpScriptStack stack, Object op1, Object op2) throws W
         GTSHelper.setValue(result, GTSHelper.tickAtIndex(gts, i), GTSHelper.locationAtIndex(gts, i), GTSHelper.elevationAtIndex(gts, i),
             operator((GTSHelper.valueAtIndex(gts, i)).toString().compareTo(op2.toString()), 0) ? GTSHelper.valueAtIndex(gts, i) : null, false);
       }
+      // Empty result should not be typed
+      if (0 == result.size()) {
+        result = new GeoTimeSerie();
```

Comment:
If op1 is bucketized, result would have been bucketized, by allocating a new series you are clearing the bucketization parameters and also the metadata. Consider using result.cloneEmpty.

Same is probably true elsewhere.

Reply:
Maybe using `setType(TYPE.UNDEFINED)` should be nicer. This would require a modification to this function as it does not check the size but the type.